basicwidget/text.go

```diff
@@ -329,13 +396,37 @@ func (t *Text) HandleInput(context *guigui.Context) guigui.HandleInputResult {
 
 	if inpututil.IsMouseButtonJustPressed(ebiten.MouseButtonLeft) {
 		if cursorPosition.In(guigui.VisibleBounds(t)) {
-			t.dragging = true
-			idx := textIndexFromPosition(textBounds, cursorPosition, t.field.Text(), face, t.lineHeight(context), t.hAlign, t.vAlign)
-			t.selectionDragStart = idx
-			guigui.Focus(t)
-			if start, end := t.field.Selection(); start != idx || end != idx {
-				t.setTextAndSelection(t.field.Text(), idx, idx, -1)
+			if ebiten.Tick()-t.lastClickTick < int64(ebiten.TPS()/2) {
+				t.clickCount++
+			} else {
+				t.clickCount = 1
 			}
+
+			switch t.clickCount {
+			case 1:
+				t.dragging = true
+				idx := textIndexFromPosition(textBounds, cursorPosition, t.field.Text(), face, t.lineHeight(context), t.hAlign, t.vAlign)
+				t.selectionDragStart = idx
+				guigui.Focus(t)
+				if start, end := t.field.Selection(); start != idx || end != idx {
+					t.setTextAndSelection(t.field.Text(), idx, idx, -1)
+				}
+			case 2:
+				t.dragging = false
```

Comment:
dragging should be true even after double clicking. Remove this line as dragging is already true.